### PR TITLE
Add macro pagetree

### PR DESCRIPTION
**Pull request by DvogelHallowelt**

### CODE REVIEW

File: tests/phpunit/data/structuredmacro-pagetree-output.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0"?>
+<xml xmlns:ac="sample_namespace" xmlns:ri="sample_second_namespace">
+	<div class="ac-pagetree">{{PageTree|reverse=false|sort=natural|startDepth=3|excerpt=true|searchBox=true|expandCollapseAll=true|content-title=ABC:SomeLinkedPage/Testpage}}</div>
+
+	<div class="ac-pagetree">{{PageTree|space-key=DEVOPS|content-title=ABC:Main Page}}</div>
+
+	<div class="ac-pagetree">{{PageTree|space-key=DEVOPS|content-title=ABC:SomeLinkedPage/Testpage}}</div>
+
+	<div class="ac-pagetree">{{PageTree|space-key=DEVOPS|content-title=ABC:SomeLinkedPage}}</div>
```
Comment (osnard) [inline tests/phpunit/data/structuredmacro-pagetree-output.xml:9]: Looks uncommon to leave a wrapping <div> when replacing macros with templates, doesn't it?

File: tests/phpunit/data/structuredmacro-pagetree-output.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+	<div class="ac-pagetree" data-params="{&quot;space-key&quot;:&quot;DEVOPS&quot;}">
+[[Category:Broken_macro/pagetree]]
```
Comment (osnard) [inline tests/phpunit/data/structuredmacro-pagetree-output.xml:12]: Can't we just replace it with
```
{{PageTree|space-key=DEVOPS}}[[Category:Broken_macro/pagetree]]
```

* If no `content-title` is set we could fall back to `{{FULLPAGENAME}}`
* If no `space-key` is set we could fall back to `{{NAMESPACE}}`

File: tests/phpunit/data/structuredmacro-pagetree-output.xml (hunk 1 above)
Comment (DvogelHallowelt) [reply tests/phpunit/data/structuredmacro-pagetree-output.xml]: removed wrapper

File: tests/phpunit/data/structuredmacro-pagetree-output.xml (hunk 2 above)
Comment (DvogelHallowelt) [reply tests/phpunit/data/structuredmacro-pagetree-output.xml]: Done but SubpageTree does not work for namespace

**Review by osnard — APPROVED** [review]

**Pull request merged**